src/compiler/types.ts

```diff
@@ -1776,9 +1781,15 @@ namespace ts {
     export interface PropertyAccessExpression extends MemberExpression, NamedDeclaration {
         kind: SyntaxKind.PropertyAccessExpression;
         expression: LeftHandSideExpression;
+        questionDotToken?: QuestionDotToken;
         name: Identifier;
     }
 
+    export interface PropertyAccessChain extends PropertyAccessExpression {
+        _optionalChainBrand: any;
+        kind: SyntaxKind.PropertyAccessExpression;

```

Comment:
It was necessary at one point in the design (each also extended an `OptionalExpression` base that also had a `kind: SyntaxKind`, but I have since removed that).